--- a/context.py
+++ b/context.py
@@ -244,16 +244,19 @@ class RequestContext(context.RequestContext):

     @property
     def trusts_auth_plugin(self):
-        if not self._trusts_auth_plugin:
-            self._trusts_auth_plugin = ks_loading.load_auth_from_conf_options(
-                cfg.CONF, TRUSTEE_CONF_GROUP, trust_id=self.trust_id)
-
-        if not self._trusts_auth_plugin:
-            LOG.error(_LE('Please add the trustee credentials you need '
-                          'to the %s section of your heat.conf file.'),
-                      TRUSTEE_CONF_GROUP)
-            raise exception.AuthorizationFailure()
-
+        cfg.CONF.import_group('keystone_authtoken',
+                              'keystonemiddleware.auth_token')
+
+        trustee_user_domain = 'default'
+        if 'user_domain_id' in cfg.CONF.keystone_authtoken:
+            trustee_user_domain = cfg.CONF.keystone_authtoken.user_domain_id
+
+        self._trusts_auth_plugin = generic.Password(
+            username=cfg.CONF.keystone_authtoken.admin_user,
+            password=cfg.CONF.keystone_authtoken.admin_password,
+            user_domain_id=trustee_user_domain,
+            auth_url=self.keystone_v3_endpoint,
+            trust_id=self.trust_id)
         return self._trusts_auth_plugin

     def _create_auth_plugin(self):

